afs-core/src/main/java/com/powsybl/afs/TaskMonitor.java

```diff
@@ -54,6 +81,17 @@ protected Task(Task other) {
             message = other.message;
             revision = other.revision;
             projectId = other.projectId;
+            cancellable = other.cancellable;
+            future = other.future;
+        }
+
+        void setFuture(Future future) {
+            this.future = future;
+            this.cancellable = future != null;
```

Comment:
Something strange here: cancellable is given as a boolean in the constructor but seems linked to the future

Reply:
yes, indeed. the future is ignored in the json serialization. it is also a package private attribute so only the taskmonitor can set it.
it should be viewed as an other way to set the value of the boolean cancellable.

Reply:
Maybe this attribute is not needed: maybe the `isCancellable` is sufficient here for serialization?

Reply:
the cancellation future has been moved out of the bean and into the localtaskmonitor